bigframes/streaming/__init__.py

```diff
@@ -13,17 +13,254 @@
+                If specified, replace the default job id for the query,
+                see job_id parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+            job_id_prefix (str, default None):
+                If specified, a job id prefix for the query, see
+                job_id_prefix parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+
+        Returns:
+            google.cloud.bigquery.QueryJob:
+                See https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJob
+                The ongoing query job can be managed using this object.
+                For example, the job can be cancelled or its error status
+                can be examined.
+        """
+        return _to_pubsub(
+            self.sql,
+            topic=topic,
+            service_account_email=service_account_email,
+            session=self._session,
+            job_id=job_id,
+            job_id_prefix=job_id_prefix,
+        )
+
+
+class StreamingDataFrame(StreamingBase):
+    __doc__ = _curate_df_doc(dataframe.DataFrame.__doc__)
+
+    # Private constructor
+    _create_key = object()
+
+    def __init__(self, df: dataframe.DataFrame, *, create_key=0):
+        if create_key is not StreamingDataFrame._create_key:
+            raise ValueError(
+                "StreamingDataFrame class shouldn't be created through constructor. Call bigframes.Session.read_gbq_table_streaming method to create."
+            )
+        self._df = df
+        self._df._disable_cache_override = True
+
+    @classmethod
+    def _from_table_df(cls, df: dataframe.DataFrame) -> StreamingDataFrame:
+        return cls(df, create_key=cls._create_key)
+
+    def __getitem__(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.__getitem__, StreamingDataFrame)(
+            *args, **kwargs
+        )
+
+    __getitem__.__doc__ = _curate_df_doc(
+        inspect.getdoc(dataframe.DataFrame.__getitem__)
+    )
+
+    def __setitem__(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.__setitem__, StreamingDataFrame)(
+            *args, **kwargs
+        )
+
+    __setitem__.__doc__ = _curate_df_doc(
+        inspect.getdoc(dataframe.DataFrame.__setitem__)
+    )
+
+    def rename(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.rename, StreamingDataFrame)(
+            *args, **kwargs
+        )
+
+    rename.__doc__ = _curate_df_doc(inspect.getdoc(dataframe.DataFrame.rename))
+
+    def __repr__(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.__repr__, StreamingDataFrame)(
+            *args, **kwargs
+        )
+
+    __repr__.__doc__ = _curate_df_doc(inspect.getdoc(dataframe.DataFrame.__repr__))
```

Comment:
This is a streaming frame, but we want to print it out just like a static frame? Could be misleading